nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

```diff
@@ -159,3 +255,139 @@
+		if(!(C.machine_stat & (BROKEN|NOPOWER|MAINT)))
+			storage_amount++
+	max_heat = initial(max_heat) + (storage_amount*storage_rate)
+	var/H = heat-cooling_amount*heat_rate
+	if(heat > 0)
+		heat = max((H),0)
+	if((weapon_state == STATE_OVERLOAD) & (heat <= (max_heat/50)))
+		weapon_state = STATE_NOTHING
+	if(weapon_state == STATE_OVERLOAD)
+		return
+	if(weapon_state == STATE_VENTING)
+		if(heat <= max_heat-ventnumber)
+			weapon_state = STATE_NOTHING
+			return
+		heat = max(((H*1.2)-H),0)
+		return
+	if(heat >= max_heat)
+		overload()
+
+
+/obj/machinery/ship_weapon/energy/proc/overload() //this is what happens when you can't control yourself
+	playsound(src, malfunction_sound, 100, 1)
+	playsound(src, overheat_sound, 100, 1)
+	do_sparks(4, FALSE, src)
+	weapon_state = STATE_OVERLOAD
+	alignment = 0
+	freq = 0
+	say("WARNING! Critical heat density, emergency venting and shutdown initiated!")
+	atmos_spawn_air("water_vapor=200;TEMP=1000")
+	heat = max_heat
+	charge = 0
+	return
+
+/obj/machinery/ship_weapon/energy/proc/handle_alignment() //this is the basic bad stuff that happens, don't fire when your gun is at 0 alignment, or it'll blow itself up
+	var/turf/detonation_turf = get_turf(src)
+	if(heat >= (3*(max_heat/4)))
+		freq -= rand(1,4)
+	if(alignment <= 75)
+		if(prob(50))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+	if(alignment <= 50)
+		if(prob(45))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+			playsound(src, malfunction_sound, 100, 1)
+		if(prob(5))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
+	if(alignment <= 25)
+		if(prob(25))
+			do_sparks(4, FALSE, src)
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+		if(prob(25))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
+		if(prob(50))
+			var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
+			var/obj/item/projectile/beam/laser/P = new(detonation_turf)
+			//Shooting Code:
+			P.range = 6
+			P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
+			P.fire()
+			freq -= rand(1,10)
+	alignment = max(alignment-(rand(0, 4)),0)
+
+
+	// dilithium crystal alignment minigame stolen from ds13
+/obj/machinery/ship_weapon/energy/screwdriver_act(mob/user, obj/item/tool)
+	. = ..()
+	if(maint_state == MSTATE_UNBOLTED)
+		.=TRUE
+		var/sound/thesound = pick('nsv13/sound/effects/computer/beep.ogg','nsv13/sound/effects/computer/beep2.ogg','nsv13/sound/effects/computer/beep3.ogg','nsv13/sound/effects/computer/beep4.ogg','nsv13/sound/effects/computer/beep5.ogg','nsv13/sound/effects/computer/beep6.ogg','nsv13/sound/effects/computer/beep7.ogg','nsv13/sound/effects/computer/beep8.ogg','nsv13/sound/effects/computer/beep9.ogg','nsv13/sound/effects/computer/beep10.ogg','nsv13/sound/effects/computer/beep11.ogg','nsv13/sound/effects/computer/beep12.ogg',)
```

Comment:
Note for a different PR: compile this into like a global list or just use a different system.
(nothing you have to worry about in your PR, you can add a comment about it if you want.)

Reply:
this was originally a list from ds13, but the sounds were in a completely different place in nsv, and I figured nobody else would use this, so there's no reason to make a list- then again it makes it more readable and stuff

Reply:
iirc we use these for console interactions as well, and you using it here proves that there may be more future use cases, so a list would be way easier